src/commands/amend.rs

```diff
@@ -0,0 +1,134 @@
+//! Amend the current commit.
+//!
+//! This command amends the HEAD commit with changes to files
+//! that are already tracked in the repo. Following the amend,
+//! the command performs a restack.
+
+use std::fmt::Write;
+use std::time::SystemTime;
+
+use eyre::Context;
+use itertools::Itertools;
+
+use crate::commands::gc::mark_commit_reachable;
+use crate::commands::restack;
+use crate::core::config::get_restack_preserve_timestamps;
+use crate::core::eventlog::{Event, EventLogDb};
+use crate::git::{AmendFastOptions, FileStatus, GitRunInfo, Repo};
+use crate::opts::MoveOptions;
+use crate::tui::Effects;
+
+/// Amends the existing HEAD commit.
+pub fn amend(
+    effects: &Effects,
+    git_run_info: &GitRunInfo,
+    move_options: &MoveOptions,
+) -> eyre::Result<isize> {
+    let now = SystemTime::now();
+    let repo = Repo::from_current_dir()?;
+    let conn = repo.get_db_conn()?;
+    let mut event_log_db = EventLogDb::new(&conn)?;
+
+    let head_oid = match repo.get_head_info()?.oid {
+        Some(oid) => oid,
+        None => eyre::bail!("No HEAD to amend to"),
+    };
+    let head_commit = repo.find_commit_or_fail(head_oid)?;
+
+    let status = repo.get_status(git_run_info)?;
```

Comment:
Calling `git status` may be an expensive operation because it has to walk the working copy, but accessing the index should be quick. Check for staged index entries and, if present, use those. Only if they're not present should you make a call to check `git status`.